pkg/externalresource/manager/service.go

```diff
@@ -54,5 +85,121 @@ func (s *Service) CreateResource(
 		return nil, st.Err()
 	}
 
+	// Note that the cache is updated AFTER the metastore is updated.
+	if _, exists := s.cache[request.GetResourceId()]; exists {
```

Comment:
As we discussed in group chat, pkg/externalresource/manager.(*Service).CreateResource doesn't have functionality of forwarding to leader, so it can't be directly embeded into servermaster.(*Server). We still need to write two place when add RPC.